ui/scripts/inlineHtml/inlineScripts.js

```diff
@@ -1035,40 +1035,69 @@ on('remove:repeating_skill',
   });
 
 // -------------- Equipment -------------
+function getLargestWeaponDefense(callback) {
+  getSectionIDs('weapon', function (ids) {
+    const defenseFields = ids.map(
+      id => `repeating_weapon_${id}_weapondefense`);
+    const countFields = ids.map(
+      id => `repeating_weapon_${id}_weaponcount`);
+    getAttrs([...defenseFields, ...countFields], function (values) {
+      const largestWeaponDefense = defenseFields.reduce((memo, defenseFieldName, i) => {
+        const defense = Number(values[defenseFieldName]);
+        const count = Number(values[countFields[i]]);
+        if (defense > 0 && count > 0) {
+          return Math.max(memo, defense);
+        }
+        return memo;
+      }, 0);
+      callback(largestWeaponDefense);
+      console.log(`Largest Weapon Defense: ${largestWeaponDefense}`);
+    });
+  });
+
+}
+
 const updateArmorValue = function () {
```

Comment:
I'd review this ignoring whitespace:
https://github.com/epic-power-rpg/roll20/pull/72/files?diff=unified&w=1

I had to indent to first retrieve all weapon defenses.